### PR TITLE
Image Banner: Add new image banner component

### LINKED ISSUE #3522: [COMPONENT] Add banner-component to extensions library
A component implementing the following design spec should be added to extensions:
https://zpl.io/ewXY1np

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [x] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [x] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [x] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [x] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by hjorte-bankdata**
## Which issue does this PR close?

#3522 

## What is the new behavior?

This adds a new image-banner component to the angular extensions library.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { argsToTemplate, type Meta, type StoryObj } from '@storybook/angular';
+import { ImageBannerComponent } from '@kirbydesign/extensions-angular/image-banner';
+
+/**
+ * TODO: Awaiting banner description from UX guidelines.
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: TODO: Add before merging. 
@henrikvoetmand do you know if we have a description?

File: libs/extensions/angular/image-banner/src/image-banner.component.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  <ng-container *ngTemplateOutlet="sharedCardContent"></ng-container>
+</kirby-card>
+
+<ng-template #sharedCardContent>
+  <div class="blur-image-wrapper">
+    <img class="blur-image" [src]="imagePath" alt="" />
+  </div>
+
+  <!-- When an external link is supplied, this anchor tag expands and fills the entire banner so users can click anywhere or focus the banner -->
+  <a *ngIf="externalLink" class="main-content-anchor" href="{{ externalLink }}" target="_blank"></a>
```
Comment (jakobe) [inline libs/extensions/angular/image-banner/src/image-banner.component.html]: Consider using same attribute binding syntax as rest of the file:
```suggestion
  <a *ngIf="externalLink" class="main-content-anchor" [href]="externalLink" target="_blank"></a>
```

File: libs/extensions/angular/image-banner/src/image-banner.component.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        <ng-container *ngIf="externalLink">
+          <!-- On large screens we also show a button-like anchor tag in addition to the entire banner anchor -->
+          <a
+            kirby-button
+            class="main-content-body-action-text"
+            [attentionLevel]="backgroundBlur === 'none' ? '3' : '2'"
+            [showIconOnly]="!actionButtonText"
+            size="sm"
+            href="{{ externalLink }}"
```
Comment (jakobe) [inline libs/extensions/angular/image-banner/src/image-banner.component.html]: Consider using same attribute binding syntax as rest of the file:
```suggestion
            [href]="externalLink"
```

File: libs/extensions/angular/image-banner/src/image-banner.component.scss [hunk 4]
```diff
@@ -0,0 +1,171 @@
+@use '@kirbydesign/core/src/scss/utils';
+@use '@kirbydesign/core/src/scss/interaction-state';
+
+$blur-radius: 60px;
+$max-content-width: 324px;
```
Comment (jakobe) [inline libs/extensions/angular/image-banner/src/image-banner.component.scss:5]: Is this just a magic number or can it be calculated somehow to give more clarity?

File: libs/extensions/angular/image-banner/src/image-banner.component.scss [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+:host(.none) {
+  .blur-image {
+    display: none;
+  }
+
+  .dismiss {
+    // on mobile we always want the button to be white as it sits on top of the image
+    @media (width < 632px) {
```
Comment (jakobe) [inline libs/extensions/angular/image-banner/src/image-banner.component.scss]: Magic number? Or do we have a breakpoint for this? 🤔

File: libs/extensions/angular/image-banner/src/image-banner.component.scss [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  top: 0;
+  left: 0;
+  width: 100%;
+  padding: utils.size('xxs');
+  box-sizing: border-box;
+  display: flex;
+  flex-direction: column;
+
+  @include utils.media('>=small') {
+    display: flex;
```
Comment (jakobe) [inline libs/extensions/angular/image-banner/src/image-banner.component.scss]: Redundant:
```suggestion
```

File: libs/extensions/angular/image-banner/src/image-banner.component.scss (hunk 5 above)
Comment (RasmusKjeldgaard) [reply libs/extensions/angular/image-banner/src/image-banner.component.scss]: There is a breakpoint, forgot to use it!

File: libs/extensions/angular/image-banner/src/image-banner.component.scss (hunk 4 above)
Comment (RasmusKjeldgaard) [reply libs/extensions/angular/image-banner/src/image-banner.component.scss:5]: Unfortunately just a magic number specific to this component :/

**Review by jakobe — DISMISSED** [review]
👌🏻

**Review by jakobe — APPROVED** [review]
👌🏻 🍂

**Pull request merged**